### ISSUE TITLE
[Provisioner] Check whether disk_size should be increased for large AMIs

### ISSUE THREAD

**Issue by infwinston** [skypilot-org/skypilot#1233]
Our default disk size is 256GB. We should check whether AMIs >256GB requires adjusting the disk size and update our document. Also, we may consider highlighting SkyPilot's support for AMI as it's not obvious for users when reading the document.

**Comment by romilbhardwaj** [comment]
disk_size indeed needs to be increased. I made a 500 GB AMI and tried provisioning it with SkyPilot without specifying a `disk_size` and got this error:
```
Task from YAML spec: custom_image.yaml
W 10-12 11:30:35 resources.py:591] image_id in resources is experimental. It only supports AWS/GCP.
I 10-12 11:30:37 optimizer.py:605] == Optimizer ==
I 10-12 11:30:37 optimizer.py:628] Estimated cost: $0.4 / hour
I 10-12 11:30:37 optimizer.py:628] 
I 10-12 11:30:37 optimizer.py:685] Considered resources (1 node):
I 10-12 11:30:37 optimizer.py:713] ------------------------------------------------------------------
I 10-12 11:30:37 optimizer.py:713]  CLOUD   INSTANCE      vCPUs   ACCELERATORS   COST ($)   CHOSEN   
I 10-12 11:30:37 optimizer.py:713] ------------------------------------------------------------------
I 10-12 11:30:37 optimizer.py:713]  AWS     m6i.2xlarge   8       -              0.38          ✔
I 10-12 11:30:37 optimizer.py:713] ------------------------------------------------------------------
I 10-12 11:30:37 optimizer.py:713]
Launching a new cluster 'myimg'. Proceed? [Y/n]: Y
I 10-12 11:30:40 cloud_vm_ray_backend.py:2705] Creating a new cluster: "myimg" [1x AWS(m6i.2xlarge, image_id='ami-03d5396bcd9a5d81e')].
I 10-12 11:30:40 cloud_vm_ray_backend.py:2705] Tip: to reuse an existing cluster, specify --cluster (-c). Run `sky status` to see existing clusters.
I 10-12 11:30:50 cloud_vm_ray_backend.py:903] To view detailed progress: tail -n100 -f /home/romilb/sky_logs/sky-2022-10-12-11-30-37-153707/provision.log
I 10-12 11:30:52 cloud_vm_ray_backend.py:1153] Launching on AWS us-west-2 (us-west-2a,us-west-2b,us-west-2c,us-west-2d)
W 10-12 11:31:08 cloud_vm_ray_backend.py:596] Got error(s) in all zones of us-west-2:
W 10-12 11:31:08 cloud_vm_ray_backend.py:598]   2022-10-12 11:31:05,267 WARN node_provider.py:459 -- create_instances: Attempt failed with An error occurred (Invalid
BlockDeviceMapping) when calling the RunInstances operation: Volume of size 256GB is smaller than snapshot 'snap-0a560e8acccc5f67e', expect size >= 500GB, retrying. 
W 10-12 11:31:08 cloud_vm_ray_backend.py:598]   2022-10-12 11:31:06,092 WARN node_provider.py:459 -- create_instances: Attempt failed with An error occurred (Invalid
BlockDeviceMapping) when calling the RunInstances operation: Volume of size 256GB is smaller than snapshot 'snap-0a560e8acccc5f67e', expect size >= 500GB, retrying. 
W 10-12 11:31:08 cloud_vm_ray_backend.py:598]   2022-10-12 11:31:06,953 WARN node_provider.py:459 -- create_instances: Attempt failed with An error occurred (Invalid
BlockDeviceMapping) when calling the RunInstances operation: Volume of size 256GB is smaller than snapshot 'snap-0a560e8acccc5f67e', expect size >= 500GB, retrying. 
W 10-12 11:31:08 cloud_vm_ray_backend.py:598]   2022-10-12 11:31:07,755 WARN node_provider.py:459 -- create_instances: Attempt failed with An error occurred (Invalid
BlockDeviceMapping) when calling the RunInstances operation: Volume of size 256GB is smaller than snapshot 'snap-0a560e8acccc5f67e', expect size >= 500GB, retrying. 
W 10-12 11:31:08 cloud_vm_ray_backend.py:598]   botocore.exceptions.ClientError: An error occurred (InvalidBlockDeviceMapping) when calling the RunInstances operatio
n: Volume of size 256GB is smaller than snapshot 'snap-0a560e8acccc5f67e', expect size >= 500GB
W 10-12 11:31:20 cloud_vm_ray_backend.py:1395] Failed to acquire resources in all regions/zones of AWS. Try changing resource requirements or use another cloud.
W 10-12 11:31:20 cloud_vm_ray_backend.py:1398] 
W 10-12 11:31:20 cloud_vm_ray_backend.py:1398] Provision failed for 1x AWS(m6i.2xlarge, image_id='ami-03d5396bcd9a5d81e'). Trying other launchable resources (if any)
.
NAME   LAUNCHED   RESOURCES            STATUS   AUTOSTOP  COMMAND                       
smoke  1 day ago  1x AWS(m6i.2xlarge)  STOPPED  -         sky launch -c smoke sky_s...

sky.exceptions.ResourcesUnavailableError: Failed to provision all possible launchable resources. Relax the task's resource requirements: 1x {AWS(None, image_id='ami-
03d5396bcd9a5d81e')}
To keep retrying until the cluster is up, use the `--retry-until-up` flag.
```

I'll add a note to the docs about this.

**Comment by romilbhardwaj** [comment]
From @Michaelvll's comment on #1237:
> I think it would be useful if SkyPilot can be smarter for this. For example, if the disk_size is not specified and the image id is specified, we can set the disk_size to be the same as the image size.
> 
> For getting aws image info: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html
> 
> For getting gcp image info: https://cloud.google.com/sdk/gcloud/reference/compute/images/list

**Comment by concretevitamin** [comment]
Is this fixed by https://github.com/skypilot-org/skypilot/pull/1508?

**Comment by Michaelvll** [comment]
Yes, this should be fixed by #1508. Closing.
